core-plugin/src/main/groovy/com/github/jrubygradle/api/gems/GemVersion.groovy

```diff
@@ -0,0 +1,573 @@
+        }
+
+        int hiCompare = compare(high, other.high)
+
+        if (hiCompare) {
+            return hiCompare
+        }
+
+        if (highBoundary != other.highBoundary) {
+            if (highBoundary == OPEN_ENDED) {
+                return 1
+            } else if (other.highBoundary == OPEN_ENDED) {
+                return -1
+            }
+            return highBoundary == INCLUSIVE ? 1 : -1
+        }
+
+        0
+    }
+
+    /**
+     * examines the version range on conflict, i.e. lower bound bigger then
+     * upper bound.
+
+     * @return boolean true if lower bound bigger then upper bound
+     */
+    boolean conflict() {
+        compare(stripNonIntegerTail(low), stripNonIntegerTail(high)) == 1
+    }
+
+    /** String of the underlying data as Ivy version range.
+     *
+     * @return Gradle Ivy version range
+     */
+    String toString() {
+        if (this == EVERYTHING) {
+            '+'
+        } else if (this == NO_VERSION) {
+            ']0,0['
+        } else if (lowBoundary == INCLUSIVE && highBoundary == INCLUSIVE && low == high) {
+            low
+        } else {
+            "${lowBoundary?.low ?: EMPTY}${low ?: EMPTY},${high ?: EMPTY}${highBoundary?.high ?: EMPTY}"
+        }
+    }
+
+    private static GemVersion parseTwiddleWakka(String singleRequirement) {
+        String base = getVersionFromRequirement(singleRequirement, TWIDDLE_WAKKA)
+        List<String> parts = base.tokenize(VERSION_SPLIT)
+        if (1 == parts) {
+            throw new GemVersionException(
+                "'${singleRequirement}' does not look like a correctly formattedGEM twiddle-wakka requirement"
+            )
+        }
+        String lastNumberPart = parts[0..-2].reverse().find {
+            it =~ ONLY_DIGITS
+        }
+        if (lastNumberPart == null) {
+            throw new GemVersionException("Cannot extract last number part from '${singleRequirement}'. " +
+                NOT_GEM_REQ)
+        }
+        int bottomAdds = 3 - parts.size()
+        if (bottomAdds < 0) {
+            bottomAdds = 0
+        }
+        try {
+            Integer nextUp = lastNumberPart.toInteger() + 1
+            String leader = parts.size() <= 2 ? EMPTY : "${parts[0..-3].join(VERSION_SPLIT)}."
+            new GemVersion(
+                INCLUSIVE,
+                "${base}${'.0' * bottomAdds}",
+                "${leader}${nextUp}.0",
+                EXCLUSIVE
+            )
+        } catch (NumberFormatException e) {
+            throw new GemVersionException("Can extract last number part from '${singleRequirement}'. " +
+                NOT_GEM_REQ, e)
+        }
+    }
+
+    @CompileDynamic
+    @SuppressWarnings('NoDef')
+    private static String getVersionFromRequirement(String gemRevision, Pattern matchPattern) {
+        def matcher = gemRevision =~ matchPattern
+        matcher[0][1]
+    }
+
+//    private GemVersion(Boolean lowInclusive, String low, String high, Boolean highInclusive) {
```

Comment:
This should go as the newer static methods to construct Gemversion is a more declaritive approach. Also this constructor did not deal with open ranges.